### PR TITLE
Llm compiler

### PR DESCRIPTION
**Pull request by ofermend**
Added LLMCompiler
Migrated to LlamaIndex 0.11.4 with pydantic 2.x
Using ToolOutput in create_rag_tool, and translating the output JSON to a string.
Bug fixes

### CODE REVIEW

File: vectara_agentic/tools.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def rag_function(*args, **kwargs) -> dict[str, Any]:
                 summary_num_results=summary_num_results,
                 summary_response_lang=summary_response_lang,
                 summary_prompt_name=vectara_summarizer,
-                vectara_query_mode=reranker,
+                reranker=reranker,
```
Comment (ofermend) [inline vectara_agentic/tools.py:185]: This is an important bugfix: the previous version did not use the right reranker

File: vectara_agentic/tools.py [hunk 2]
```diff
@@ -143,6 +144,8 @@ def create_rag_tool(
             mmr_diversity_bias (float, optional): MMR diversity bias.
             include_citations (bool, optional): Whether to include citations in the response.
                 If True, uses markdown vectara citations that requires the Vectara scale plan.
+            fcs_threshold (float, optional): a threshold for factual consistency. 
+                Tool notifies caller it cannot respond if FCS it too high
```
Comment (david-oplatka) [inline vectara_agentic/tools.py:148]: Wouldn't this notify if FCS is too low, not too high?

File: vectara_agentic/tools.py [hunk 3]
```diff
@@ -199,31 +202,45 @@ def rag_function(*args, **kwargs) -> dict[str, Any]:
...
                 "response": response.response,
-                "citation_metadata": citation_metadata,
-                "factual_consistency": (
-                    response.metadata["fcs"] if "fcs" in response.metadata else 0.0
-                ),
+                "references_metadata": citation_metadata,
             }
-            return res
+
+            print(f"DEBUG res={res}")
```
Comment (david-oplatka) [inline vectara_agentic/tools.py:222]: May want to remove this

File: vectara_agentic/tools.py (hunk 2 above)
Comment (ofermend) [reply vectara_agentic/tools.py:148]: yes, correct. Updated.

File: vectara_agentic/tools.py (hunk 3 above)
Comment (ofermend) [reply vectara_agentic/tools.py]: yeah

**Pull request merged**